src/coreclr/jit/gentree.h

```diff
@@ -7661,7 +7661,12 @@ inline bool GenTree::IsSIMDZero() const
 //
 inline bool GenTree::IsFloatPositiveZero() const
 {
-    return !(IsCnsNonZeroFltOrDbl());
+    if (OperGet() == GT_CNS_DBL)
```

Comment:
This is a bug fix. Originally, this function would return 'true' if the oper was not `GT_CNS_DBL`.